File: src/main/java/com/chatsummary/bot/model/DailySummary.java
package com.chatsummary.bot.model;

import java.time.Instant;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "daily_summaries")
public record DailySummary(
        @Id String id,
        @Indexed long chatId,
        String text,
        @Indexed Instant timestamp
) {

    public DailySummary {
        timestamp = timestamp == null ? Instant.now() : timestamp;
    }

    public DailySummary(long chatId, String text) {
        this(null, chatId, text, Instant.now());
    }
}

File: src/main/kotlin/com/chatsummary/bot/model/DailySummary.kt
package com.chatsummary.bot.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.data.mongodb.core.index.Indexed
import java.time.Instant

@Document(collection = "daily_summaries")
data class DailySummary(
    @Id
    val id: String? = null,

    @Indexed
    val chatId: Long,

    val text: String,

    @Indexed
    val timestamp: Instant = Instant.now()
)
